firestore.rules

```diff
@@ -0,0 +1,30 @@
+rules_version = '2';
+service cloud.firestore {
+  match /databases/{database}/documents {
+  
+    match /userInfo/{document=**} {
+   		allow read, write;
+    }
+
+    match /users/{document=**} {
+    	allow read, write;
```

Comment:
These should be more restricted. According to this rule any user can update any users' info. This should be something like 
```
"users": {
      "$uid": {
        ".write": "$uid === auth.uid"
    }
}
```
where this will only grant edit access to the relevant use who has the uid. Same as this, rules should be there for each possible unit data(which means if there are two collections starts with users -> ex: users/uid, users/profile, each case should have their own rule if the rule is different, if the collection shares a common rule, we can use regex method that you have already used in here). Update the rules accordingly. And when committing, let's create a separate directory to store the rules instead adding them to the root directory. I suggest directory -> firebase_rules, if there is more convenient name, let's use that.

References: https://firebase.google.com/docs/database/security

Reply:
Since there is no much work we have to do here(I think we are almost done, correct me if I'm wrong), let's focus on this particular case and writing more test cases.

Reply:
Done! I have created different segments for queries and users. Updated file path with `firebase_rules` directory.